server_modules/scores.js

```diff
@@ -0,0 +1,49 @@
+var utils = require('./utils');
+var fileSystem = require('./file_system');
+
+function filterPublished(score) {
+    return score.published;
+}
+
+function reduceToMap(key) {
+    return function(arr) {
+        return arr.reduce(function(map,record) {
+            map[record[key]] = record;
+            return map;
+        },{});
+    }
+}
+
+exports.route = function(app) {
+
+    //get all, grouped by round
+    app.get('/scores/',function(req,res) {
+        Q.all([
+            fileSystem.parseDataFile('scores.json'),

```

Comment:
I just read it in more details... It just add a try-catch mechanism around the JSON parsing, using the JSON.parse function, which I guess I get, since it's server side code. I'll change the name to be clearer